bin/queuePopulator.js

```diff
@@ -59,6 +66,19 @@ async.waterfall([
         });
         done();
     },
+    done => {
+        healthServer.onReadyCheck(log => {
+            const state = queuePopulator.zkStatus();
+            if (state.code === zookeeper.State.SYNC_CONNECTED.code) {
+                return true;
+            }
+            log.error(`Zookeeper is not connected! ${state}`);
```

Comment:
I have tried to debug backbeat and our other services multiple times where the liveness check that exists fails with no indication as to which part of the check actually failed. It is just some magic that you have to know when a liveness fails here it is probably related to X service. I think it would be much better to return some sort of error response indicating which check failed. The current HealthProbeServer doesn't really allow us to do that though. I did a quick redesign locally to see how we might go about it though and it does not seem too bad. 

See the example here: https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#define-a-liveness-http-request
I think it would be especially important when we get multiple checks.